### PR TITLE
 zfs: 2.0.0 -> 2.0.1 

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -190,22 +185,13 @@ in {
...
-
-    # this package should point to a version / git revision compatible with the latest kernel release
-    version = "2.0.0";
-
-    sha256 = "1kriz6pg8wj98izvjc60wp23lgcp4k3mzhpkgj74np73rzgy6v8r";
-
-    extraPatches = [ linux-rt-patch ];
-  };
+  # this package should point to a version / git revision compatible with the latest kernel release
+  zfsUnstable = zfsStable;
```
Comment (Mic92) [inline pkgs/os-specific/linux/zfs/default.nix:196]: I actually would keep the old structure and just make it the same as `zfsStable`. We likely need this in future again.

Comment (Atemu) [reply pkgs/os-specific/linux/zfs/default.nix:196]: You could just copy pasta the structure of stable above.

Comment (hmenke) [reply pkgs/os-specific/linux/zfs/default.nix:196]: I agree with Mic92. If we add and remove this block each time a release candidate comes out, the diffs become quite messy and hard to understand.

**Comment by Atemu** [comment]
Result of `nixpkgs-review pr 108646` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>gnome3.gnome-boxes</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libvmi</li>
    <li>linuxPackages-libre.zfs (linuxPackages-libre.zfsUnstable)</li>
    <li>nixops</li>
    <li>nixopsUnstable</li>
    <li>nixops_1_6_1</li>
    <li>python37Packages.guestfs</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>samba4Full</li>
    <li>vagrant</li>
  </ul>
</details>
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>grub2</li>
    <li>grub2_efi</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>libceph</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt_5_9_0</li>
    <li>linuxPackages.zfs (linuxPackages.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsUnstable)</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsUnstable)</li>
    <li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsUnstable)</li>
    <li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsUnstable)</li>
    <li>linuxPackages_5_10.zfs (linuxPackages_5_10.zfsUnstable)</li>
    <li>linuxPackages_5_9.zfs (linuxPackages_5_9.zfsUnstable)</li>
    <li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest-libre.zfs (linuxPackages_latest-libre.zfsUnstable)</li>
    <li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0.zfs (linuxPackages_latest_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0_hardened.zfs (linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsUnstable)</li>
    <li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0.zfs (linuxPackages_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0_hardened.zfs (linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_zen.zfs (linuxPackages_zen.zfsUnstable)</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>os-prober</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python37Packages.libvirt</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.libvirt</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_2_5.ruby-libvirt</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>sanoid</li>
    <li>terraform-full</li>
    <li>terraform-provider-libvirt (terraform-providers.libvirt)</li>
    <li>virtmanager (virt-manager)</li>
    <li>virtmanager-qt (virt-manager-qt)</li>
    <li>virt-top</li>
    <li>virtviewer (virt-viewer)</li>
    <li>virtlyst</li>
    <li>zfs (zfsUnstable)</li>
    <li>zfstools</li>
  </ul>
</details>

**Review by hmenke — APPROVED** [review]

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 108646` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>gnome3.gnome-boxes</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libvmi</li>
    <li>linuxPackages-libre.zfs (linuxPackages-libre.zfsUnstable)</li>
    <li>nixops</li>
    <li>nixopsUnstable</li>
    <li>nixops_1_6_1</li>
    <li>python37Packages.guestfs</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>samba4Full</li>
    <li>vagrant</li>
  </ul>
</details>
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>grub2</li>
    <li>grub2_efi</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>libceph</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt_5_9_0</li>
    <li>linuxPackages.zfs (linuxPackages.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsUnstable)</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsUnstable)</li>
    <li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsUnstable)</li>
    <li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsUnstable)</li>
    <li>linuxPackages_5_10.zfs (linuxPackages_5_10.zfsUnstable)</li>
    <li>linuxPackages_5_9.zfs (linuxPackages_5_9.zfsUnstable)</li>
    <li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest-libre.zfs (linuxPackages_latest-libre.zfsUnstable)</li>
    <li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0.zfs (linuxPackages_latest_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0_hardened.zfs (linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsUnstable)</li>
    <li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0.zfs (linuxPackages_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0_hardened.zfs (linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_zen.zfs (linuxPackages_zen.zfsUnstable)</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>os-prober</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python37Packages.libvirt</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.libvirt</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_2_5.ruby-libvirt</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>sanoid</li>
    <li>terraform-full</li>
    <li>terraform-provider-libvirt (terraform-providers.libvirt)</li>
    <li>virtmanager (virt-manager)</li>
    <li>virtmanager-qt (virt-manager-qt)</li>
    <li>virt-top</li>
    <li>virtviewer (virt-viewer)</li>
    <li>virtlyst</li>
    <li>zfs (zfsUnstable)</li>
    <li>zfstools</li>
  </ul>
</details>

**Comment by jonringer** [comment]
@GrahamcOfBorg test zfs.stable

**Comment by jonringer** [comment]
failing builds are broken in master:
```
12 packages failed to build:
gnome3.gnome-boxes libguestfs libvmi linuxPackages-libre.zfs nixops nixopsUnstable nixops_1_6_1 python37Packages.guestfs python38Packages.guestfs python39Packages.guestfs sambaFull vagrant

55 packages built:
ceph ceph-client collectd collectd-data docker-machine-kvm docker-machine-kvm2 easysnap grub2 grub2_efi grub2_pvgrub_image grub2_xen haskellPackages.libvirt-hs haskellPackages.libzfs libceph libvirt libvirt-glib libvirt_5_9_0 linuxPackages.zfs linuxPackages_4_14.zfs linuxPackages_4_19.zfs linuxPackages_4_4.zfs linuxPackages_4_9.zfs linuxPackages_5_10.zfs linuxPackages_5_9.zfs linuxPackages_hardened.zfs linuxPackages_latest-libre.zfs linuxPackages_latest_hardened.zfs linuxPackages_latest_xen_dom0.zfs linuxPackages_latest_xen_dom0_hardened.zfs linuxPackages_lqx.zfs linuxPackages_testing_bcachefs.zfs linuxPackages_xen_dom0.zfs linuxPackages_xen_dom0_hardened.zfs linuxPackages_zen.zfs minikube minishift os-prober perl530Packages.SysVirt perl532Packages.SysVirt python37Packages.libvirt python38Packages.libvirt python39Packages.libvirt rubyPackages.ruby-libvirt rubyPackages_2_5.ruby-libvirt rubyPackages_2_7.ruby-libvirt sanoid terraform-full terraform-providers.libvirt virt-manager virt-manager-qt virt-top virt-viewer virtlyst zfs zfstools
```

**Pull request merged**